integration_tests/tests/query_tests/test_queries.py

```diff
@@ -503,6 +537,60 @@ def test_run_query(query_kind, params, universal_access_token, flowapi_url):
     assert len(result_dataframe) > 0
 
 
+@pytest.mark.parametrize(
+    "query_kind, params",
+    [
+        (
+            "joined_spatial_aggregate",
+            {
+                "locations": {
+                    "query_kind": "daily_location",
+                    "date": "2016-01-01",
+                    "aggregation_unit": "admin3",
+                    "method": "last",
+                },
+                "metric": {
+                    "query_kind": "topup_balance",
+                    "start_date": "2016-01-01",
+                    "end_date": "2016-01-02",
+                    "statistic": "avg",
+                },
+                "method": "dist",
+            },
+        ),
+        (
+            "joined_spatial_aggregate",
+            {
+                "locations": {
+                    "query_kind": "daily_location",
+                    "date": "2016-01-01",
+                    "aggregation_unit": "admin3",
+                    "method": "last",
+                },
+                "metric": {
+                    "query_kind": "handset",
+                    "start_date": "2016-01-01",
+                    "end_date": "2016-01-02",
+                    "characteristic": "hnd_type",
+                    "method": "last",
+                },
+                "method": "avg",
+            },
+        ),
+    ],
+)
+def test_fail_query_incorret_parameters(
+    query_kind, params, universal_access_token, flowapi_url
+):
+    """
+    Test that queries fail with incorrect parameters.
+    """
+    query_spec = getattr(flowclient, query_kind)(**params)
+    con = flowclient.Connection(url=flowapi_url, token=universal_access_token)
+    with pytest.raises(Exception):
```

Comment:
Great to have this test for failure with incorrect parameters, but I suggest making the assertion a bit more specific:
- Capture the specific type of exception you expect (e.g. `FlowclientConnectionError`?) rather than a generic one.
- Add a `match="..."` argument to `pytest.raises(...)` so that we know the error message contains the expected test.

Otherwise it's quite possible that the test passes because _some_ exception is raised, but it's actually hiding a different error which we don't get to see. (Which isn't the case here but it's definitely happened to me in the past.)

I suggest something along the lines of:
```
with pytest.raises(FlowclientConnectionError, match="Must be one of: avg, max, min, median, mode, stddev, variance."):
    ...
```

Reply:
FWIW, the value of the `match` parameter may (or may not) need tweaking now that #1056 is merged.